compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -418,7 +398,8 @@ pub fn const_validate_mplace<'mir, 'tcx>(
                 CtfeValidationMode::Const { allow_immutable_unsafe_cell: !inner }
             }
         };
-        ecx.const_validate_operand(&mplace.into(), path, &mut ref_tracking, mode)?;
+        ecx.const_validate_operand(&mplace.into(), path, &mut ref_tracking, mode)
+            .map_err(|error| const_report_error(&ecx, error, alloc_id))?;
```

Comment:
`const_report_error` seems to be specifically for reporting *validation* failures -- might be better to rename it to `report_validation_error` and make it private?

Also would be good to explain here why we don't just forward the `InterpError`: to give a more targeted error when final validation fails.